projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.ts

```diff
@@ -142,74 +116,105 @@ export class ValidationMessagesComponent
     );
   }
 
-  ngOnInit(): void {
-    this.readFormControlByControlName();
-  }
-
-  ngAfterContentInit(): void {
-    this.readFormControlFromHost();
-
-    this.control.valueChanges
-      .pipe(takeUntil(this.ngUnsubscribe))
-      .subscribe(() => this.cd.markForCheck());
-  }
-
-  ngOnDestroy(): void {
-    this.ngUnsubscribe.next();
-    this.ngUnsubscribe.complete();
-  }
-
-  ngDoCheck(): void {
-    if (
-      this.control &&
-      ((this.control.invalid && this.control.touched) ||
-        (!this.control.invalid && this.shownErrors.length > 0))
-    ) {
-      this.updateErrorMessages();
-      this.cd.markForCheck();
-    }
-  }
-
-  private updateErrorMessages(): void {
-    this.shownErrors = [];
+  private _updateErrorMessages(): void {
```

Comment:
I'm wondering if could make `shownErrors` signal as `computed` and move the whole logic there